workspace/src/app/views/shared/TaskActivityOverview/TaskActivityOverview.tsx

```diff
@@ -105,6 +113,24 @@ export function TaskActivityOverview({ projectId, taskId }: IProps) {
         );
     };
 
+    const statusMapDependency = Array.from(activityStatusMap.values())
+        .map((v) => `${v.lastUpdateTime}-${v.concreteStatus}-${v.progress}`)
+        .join("|");
+
+    React.useEffect(() => {
+        const currentlyRunningActivity = Array.from(activityStatusMap.values()).find((a) => a.isRunning);
+        const activity = cacheActivities[cursor.current];
```

Comment:
It would be better to start all activities at once, since there can be multiple long-running activities whose execution times will add up.